autoload/go/path.vim

```diff
@@ -139,26 +139,29 @@ function! go#path#CheckBinPath(binpath)
     " remove whitespaces if user applied something like 'goimports   '
     let binpath = substitute(a:binpath, '^\s*\(.\{-}\)\s*$', '\1', '')
 
+    " check if we have an appropriate bin_path
+    let go_bin_path = go#path#BinPath()
```

Comment:
Can you please handle the case when this is empty (it can be). You should handle it here, not below.